FWCore/Modules/src/TimeStudyModules.cc

```diff
@@ -227,12 +227,13 @@ namespace timestudy {
         return true;
       }
       //every running stream is now waiting
-      return waitingStreams_.size() == activeStreams_;
+      return !waitingStreams_.empty() and waitingStreams_.size() == activeStreams_;
     }
 
     void threadWork() {
       while (not stopProcessing_.load()) {
         std::vector<int> streamsToProcess;
+        streamsToProcess.reserve(waitingStreams_.capacity());
```

Comment:
This is a race condition. In order to interact with `waitingStreams_` the `mutex_` must be taken.